File: Arrays & Hashing/1700. Number of Students Unable to Eat Lunch/1700.cpp
#pragma GCC optimize('0fast','unroll-loops')

class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        deque<int> student(students.begin(),students.end());
        int idx=0;
        int visit=0;

        while(!student.empty() && visit!=student.size()){
            if(sandwiches[idx]==student.front()){
                idx++;
                visit=0;
                student.pop_front();
            } else{
                visit++;
                int st=student.front();
                student.pop_front();
                student.push_back(st);
            }
        }

        return student.size();
    }
};

File: Arrays & Hashing/1700. Number of Students Unable to Eat Lunch/1700.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        student=deque()
        sandwich=deque()
        for i in range(len(students)):
            student.append((students[i],i))
            sandwich.append(sandwiches[i])

        visited=set()
        
        while sandwich:
            if (student[0][1],len(sandwich)) in visited:
                return len(student)
            if sandwich[0]==student[0][0]:
                sandwich.popleft()
                student.popleft()
            else:
                visited.add((student[0][1],len(sandwich)))
                st,idx=student.popleft()
                student.append((st,idx))
        return 0
